redbeams/src/main/java/com/sequenceiq/redbeams/controller/v4/databaseserver/DatabaseServerV4Controller.java

```diff
@@ -217,4 +222,11 @@ public void start(@TenantAwareParam @ResourceCrn @ValidCrn(resource = CrnResourc
     public void stop(@TenantAwareParam @ResourceCrn @ValidCrn(resource = CrnResourceDescriptor.DATABASE_SERVER) @NotNull String crn) {
         redbeamsStopService.stopDatabaseServer(crn);
     }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.UPGRADE_DATABASE_SERVER)
+    public void upgrade(@TenantAwareParam @ResourceCrn @ValidCrn(resource = CrnResourceDescriptor.DATABASE_SERVER) @NotNull String crn,
```

Comment:
who will call this? if only another CDP service, then this should be `@InternalOnly` endpoint. If there will be CDP CLI command or UI button for this, then this will be ok, but only for internal call, I would not introduce a new permission, since usually redbeams is only called by other services afaik

Reply:
@horadla23 I was following the authorization pattern in redbeams controller. Most operations, like stop / start, delete do have authorization even though those API methods are not called directly (exception is the deprecated dp cli) from UI / cdp-cli, only via other services.

Reply:
in case of internal we would not need new permission in ums, we have tons of permission already

Reply:
Ok, I make this one internal then, and add a jira to our epic to make all the other ones internal as well.

Reply:
jira id: https://jira.cloudera.com/browse/CB-17385